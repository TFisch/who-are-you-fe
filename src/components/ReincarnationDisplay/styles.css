.reincarnation-display {
  border: 1px solid black;
  display: flex;
  flex-direction: column;
  height: 23em;
  justify-content: space-between;
  width: 40%; }

.comment {
  height: 5em;
  width: 40%; }

.comment-wrapper {
  margin-left: 1em;
  margin-right: 1em; }

.bottom-wrapper {
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding-bottom: 2em; }
