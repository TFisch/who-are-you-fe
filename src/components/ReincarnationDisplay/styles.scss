.reincarnation-display {
  border: 1px solid black;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 23em;
  width: 40%;
  margin: 2rem auto auto auto;
}

.comment {
  height: 5em;
  width: 40%;
}

.comment-wrapper {
  margin-left: 1em;
  margin-right: 1em;
}

.bottom-wrapper {
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding-bottom: 2em;
}

.past-life-message {
  text-align: center;
  margin: auto;
}

.no-reincarnation {
  text-align: center;
  margin: auto;
}
