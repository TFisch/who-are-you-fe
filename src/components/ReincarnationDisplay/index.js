import React, { Component } from 'react';
import Submit from '../common/Submit';
import './styles.scss';


class ReincarnationDisplay extends Component {
  constructor() {
    super()
    this.state = {
      deathMatch: {}
    }
  }

  componentDidMount = () => {
    this.setState({ deathMatch: this.props.deathsByDate })
  }


  handleSubmit = (e) => {
    console.log(e);
  }

  render() {
    const name = Object.values(this.props.deathsByDate[0]);
    return (
      <article className="reincarnation-display">
        <h2>{this.props.username}! You Were {name[1]} In Your Past Life!</h2>
        <div className="comment-wrapper">
          <label>Leave a comment:</label>
          <div className="bottom-wrapper">
            <textarea className="comment" name="comment">
            </textarea>
            <Submit buttonText="SAVE" handleSubmit={this.handleSubmit} />
          </div>
        </div>
      </article>
    )
  }
}

export default ReincarnationDisplay;