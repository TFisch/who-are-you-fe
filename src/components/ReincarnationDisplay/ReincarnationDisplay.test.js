import React from 'react';
import { shallow } from 'enzyme';
import ReincarnationDisplay from './'

describe('ReincarnationDisplay', () => {
  let wrapper;

  beforeEach(() => {
    let mockProps = {
      buttonText: "mock"
    }

    wrapper = shallow(<ReincarnationDisplay props={mockProps}
    />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});