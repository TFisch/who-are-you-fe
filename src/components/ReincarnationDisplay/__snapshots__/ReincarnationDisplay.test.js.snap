// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReincarnationDisplay should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReincarnationDisplay
    props={
      Object {
        "buttonText": "mock",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Name! You Were DEAD PERSON in Your Past Life!
        </h2>,
        <div
          className="comment-wrapper"
        >
          <label
            for="comment"
          >
            Leave a comment:
          </label>
          <div
            className="bottom-wrapper"
          >
            <textarea
              class="comment"
              name="comment"
            />
            <Submit
              buttonText="SAVE"
              handleSubmit={[Function]}
            />
          </div>
        </div>,
      ],
      "className": "reincarnation-display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Name! You Were DEAD PERSON in Your Past Life!",
        },
        "ref": null,
        "rendered": "Name! You Were DEAD PERSON in Your Past Life!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              for="comment"
            >
              Leave a comment:
            </label>,
            <div
              className="bottom-wrapper"
            >
              <textarea
                class="comment"
                name="comment"
              />
              <Submit
                buttonText="SAVE"
                handleSubmit={[Function]}
              />
            </div>,
          ],
          "className": "comment-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Leave a comment:",
              "for": "comment",
            },
            "ref": null,
            "rendered": "Leave a comment:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <textarea
                  class="comment"
                  name="comment"
                />,
                <Submit
                  buttonText="SAVE"
                  handleSubmit={[Function]}
                />,
              ],
              "className": "bottom-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "class": "comment",
                  "name": "comment",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "buttonText": "SAVE",
                  "handleSubmit": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Name! You Were DEAD PERSON in Your Past Life!
          </h2>,
          <div
            className="comment-wrapper"
          >
            <label
              for="comment"
            >
              Leave a comment:
            </label>
            <div
              className="bottom-wrapper"
            >
              <textarea
                class="comment"
                name="comment"
              />
              <Submit
                buttonText="SAVE"
                handleSubmit={[Function]}
              />
            </div>
          </div>,
        ],
        "className": "reincarnation-display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Name! You Were DEAD PERSON in Your Past Life!",
          },
          "ref": null,
          "rendered": "Name! You Were DEAD PERSON in Your Past Life!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                for="comment"
              >
                Leave a comment:
              </label>,
              <div
                className="bottom-wrapper"
              >
                <textarea
                  class="comment"
                  name="comment"
                />
                <Submit
                  buttonText="SAVE"
                  handleSubmit={[Function]}
                />
              </div>,
            ],
            "className": "comment-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Leave a comment:",
                "for": "comment",
              },
              "ref": null,
              "rendered": "Leave a comment:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <textarea
                    class="comment"
                    name="comment"
                  />,
                  <Submit
                    buttonText="SAVE"
                    handleSubmit={[Function]}
                  />,
                ],
                "className": "bottom-wrapper",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "class": "comment",
                    "name": "comment",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "buttonText": "SAVE",
                    "handleSubmit": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
